lib/date/converter.go

```diff
@@ -0,0 +1,17 @@
+package date
+
+import "time"
+
+const (
+	// RFC3339 time layout
+	RFC3339Nano = "2006-01-02T15:04:05.999999999Z07:00"
+)
+
+// ToUnixNano converts a RFC3339
+func ToUnixNano(dateStr string) (int64, error) {
+	t, err := time.Parse(RFC3339Nano, dateStr)
```

Comment:
What's a good way to detect date format from user input? I would prefer to add additional layouts here.

Reply:
Maybe use a library? https://github.com/araddon/dateparse
Not ideal though. New dependency.

Reply:
Yeah, I guess better to stick to standard date time, else it can cause confusions and different queries can return different results.

Reply:
So the date time format is defined by us, which we are using for validation purposes. Let's stick to that format.

Also, this format needs to be in the schema package close to the DateTime data type so that we know that this format is what we are proposing to the end user for this data type.  This function can still stay here and you can accept the format which will be defined in the schema package.